.github/workflows/pull-request-checks.yml

```diff
@@ -92,12 +89,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Install nightly toolchain
-        id: toolchain
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: nightly
-      - run: rustup override set ${{steps.toolchain.outputs.name}}
+      - name: Install nightly
+        uses: dtolnay/rust-toolchain@nightly
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Run cargo-udeps
+        run: |
+          cargo install cargo-udeps
+          cargo udeps --all-features --all-targets
```

Comment:
Good idea to include all features and targets and to detect unused  dev dependencies!